lib/pbench/server/templates.py

```diff
@@ -292,9 +331,16 @@ def __init__(
             settings: JSON index settings. Defaults to None.
             skeleton: A versionless skeleton mappings file to merge with the tool
                 mappings.
-            tool: Tool mapping files stitch together mapping JSON from the
-                main and skeleton templates using a parsed "tool" name.
-        """
+            tool: The key in the index_patterns dict differs from the actual
+                tool name: e.g., all "tool" documents share the "tool-data" tool
+                name.
+        """
+        # NOTE: we really don't want prefix to be None, but it is in some CLI
+        # unit tests and it's not worth the churn to fix that.
+        if mappings is None or settings is None:
+            raise TemplateError(
+                f"Missing parameters MAPPINGS={mappings}, SETTINGS={settings}"
+            )
```

Comment:
The whole `None` thing was because I started out with chained `TemplateFile` objects where you could build a "base" `TemplateFile` with just `settings`, for example, that would then be used by a series of `TemplateFile` objects that referred to it as a `base`. When I removed that obfuscation, some obfuscation droppings remained which I'd simply grown used to stepping over...